aleph_export/harvest_aleph_marc.py

```diff
@@ -0,0 +1,88 @@
+import time
+import json
+import os
+from transform_marc_json import TransformMarcJson
+from dependencies.pymarc import MARCReader
+from dependencies.sentry_sdk import capture_exception
+import dependencies.requests
+from dependencies.pipelineutilities.search_files import crawl_available_files
+from dependencies.pipelineutilities.s3_helpers import write_s3_file
+
+
+class HarvestAlephMarc():
+    """ This performs all Marc-related processing """
+    def __init__(self, config, event, marc_records_url):
+        self.config = config
+        self.marc_records_url = marc_records_url
+        self.start_time = time.time()
+        self.event = event
+        self.hash_of_available_files = {}
+        if not self.event['local']:
+            self.hash_of_available_files = crawl_available_files(self.config)
+            # with open(os.path.join('test', 'hash_of_available_files.json'), "w") as file1:
+            #     file1.write(json.dumps(self.hash_of_available_files, indent=2, default=str))
+        self.temporary_local_path = '/tmp'
+        self.marc_records_stream = self._open_marc_records_stream()
+
+    def _open_marc_records_stream(self):
+        """ Return marc records from URL."""
+        marc_records_stream = ""
+        url = self.marc_records_url
+        try:
+            r = dependencies.requests.get(url, stream=True)
+            if r.status_code == 200:
+                marc_records_stream = r.raw
+        except ConnectionRefusedError:
+            capture_exception('Connection refused on url ' + url)
+        except:  # noqa E722 - intentionally ignore warning about bare except
+            capture_exception('Error caught trying to process url ' + url)
+        return marc_records_stream
+
+    def process_marc_records_from_stream(self, test_mode_flag=False):
+        marc_reader = MARCReader(self.marc_records_stream)
+        transform_marc_json_class = TransformMarcJson(self.config["csv-field-names"], self.hash_of_available_files)
+        processed_records_count = 0
+        for marc_record in marc_reader:
+            marc_record_as_json = json.loads(marc_record.as_json())
+            # for field in marc_record_as_json['fields']:
```

Comment:
Should these comments be removed as old code?

Reply:
I intentionally left this code in case we need to export to re-create files for samples if the source marc changes.  In fact, I used this today, now that Peggy has added the subject uri to marc records.